pandas/tests/tools/test_to_datetime.py

```diff
@@ -249,7 +246,7 @@ def test_to_datetime_format_integer(self, cache):
             # valid date, length == 8
             [20121030, datetime(2012, 10, 30)],
             # short valid date, length == 6
-            [199934, datetime(1999, 3, 4)],
+            [199934, 199934],
```

Comment:
199934 doesn't match the given format (%Y%m%d), so it shouldn't parse. In particular, with `errors='ignore'`, it should return the input

This test looked wrong to begin with

Reply:
Gonna block myself here, as both `pyarrow` and `datetime` would parse this:
```python
In [5]: pa.compute.strptime('199934', format='%Y%m%d', unit='ns')
Out[5]: <pyarrow.TimestampScalar: Timestamp('1999-03-04 00:00:00')>

In [6]: dt.datetime.strptime('199934', '%Y%m%d')
Out[6]: datetime.datetime(1999, 3, 4, 0, 0)
```

This'll need handling in

https://github.com/pandas-dev/pandas/blob/a37b78d534cd25e21d2ab64318ceae808df46c89/pandas/_libs/tslibs/src/datetime/np_datetime_strings.c#L101-L105

Reply:
solved this by just not using the ISO8601 fast path. better to take a slight performance hit than to have bugs IMO